### PR TITLE
Generatorify

### PR DESCRIPTION
**Pull request by incompl**

### CODE REVIEW

File: test/videojs.ads.test.js [hunk 1]
```diff
@@ -936,7 +936,15 @@ QUnit.test('ended event is sent with postroll', function(assert) {
 
   var ended = sinon.spy();
```
Comment (ldayananda) [inline test/videojs.ads.test.js]: Curious why this test needed to be changed?

Comment (incompl) [inline test/videojs.ads.test.js]: In Phantom you can set ended on a video element, in a real browser it's read-only, so instead I mock the element completely.

File: migrating-to-4.0.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Migrating to videojs-contrib-ads 4.0.0
+
+Version 4 of videojs-contrib-ads is mainly about improving code quality. Functional
+changes are in support of that effort.
+
+## Migration
+
+* A `playing` event is no longer sent before a preroll ad. This is because a `playing`
+event should only be sent when content actually plays. If your integration relies on
+this event being fired at this time, consider another solution.
```
Comment (incompl) [inline migrating-to-4.0.md]: Will add

File: README.md [hunk 3]
```diff
@@ -284,41 +283,28 @@ if (player.ads.state  === 'ad-playback') {
...
+The ads plugin is designed to be built with `npm`.
 
 If you don't already have `npm`, then download and install [Node.js](http://nodejs.org/) (which comes with npm).
-Then you can install the build tool [grunt](http://gruntjs.com/):
-
-```sh
-$ npm install -g grunt
-```
 
 With grunt ready, you can download the ads plugin's build-time dependencies and then build the ads plugin.
```
Comment (incompl) [inline README.md]: Fixing

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,31 +1,30 @@
-# Contributing
+# CONTRIBUTING
 
-## Important notes
```
Comment (incompl) [inline CONTRIBUTING.md]: I'd rather use the stock CONTRIBUTING file, I'm fine with leaving out the common sense stuff.

File: test/videojs.ads.test.js (hunk 1 above)
Comment (ldayananda) [inline test/videojs.ads.test.js]: That makes sense

**Comment by ldayananda** [comment]
LGTM

**Pull request merged**